### PR TITLE
Add last modified file (gm)

### PR DESCRIPTION
**Pull request by pickfire**
cc @EpocSquadron 

This will change `gm` (goto middle) to `gc` but now it matches kakoune behavior and our `zc`.

Based on https://github.com/mawww/kakoune/pull/3814

### CODE REVIEW

File: book/src/keymap.md [hunk 1]
```diff
@@ -156,13 +156,14 @@ Jumps to various locations.
 | `l`   | Go to the end of the line                        | `goto_line_end`            |
 | `s`   | Go to first non-whitespace character of the line | `goto_first_nonwhitespace` |
 | `t`   | Go to the top of the screen                      | `goto_window_top`          |
-| `m`   | Go to the middle of the screen                   | `goto_window_middle`       |
+| `c`   | Go to the middle of the screen                   | `goto_window_middle`       |
```
Comment (sudormrfbin) [inline book/src/keymap.md:159]: Maybe we should rename it to `goto_window_center` to be consistent with the keybind ?

Comment (pickfire) [reply book/src/keymap.md:159]: Sure.

File: helix-view/src/view.rs [hunk 2]
```diff
@@ -65,6 +65,9 @@ pub struct View {
     pub jumps: JumpList,
     /// the last accessed file before the current one
     pub last_accessed_doc: Option<DocumentId>,
+    /// the last modified file before the current one
+    pub last_modified_doc1: Option<DocumentId>,
+    pub last_modified_doc2: Option<DocumentId>,
```
Comment (archseer) [inline helix-view/src/view.rs:70]: Why not `last_modified_docs: [Option<DocumentId>; 2]`?

I also don't see an explanation on why we need two fields

Comment (pickfire) [reply helix-view/src/view.rs:70]: Sorry, didn't put a comment on that. Yeah, this is good as well.

File: helix-term/src/keymap.rs [hunk 3]
```diff
@@ -520,9 +520,10 @@ impl Default for Keymaps {
                 "r" => goto_reference,
                 "i" => goto_implementation,
                 "t" => goto_window_top,
-                "m" => goto_window_middle,
+                "c" => goto_window_middle,
```
Comment (archseer) [inline helix-term/src/keymap.rs:523]: I rejected changing this keybind before :P why not find a different keymap for `goto_last_modified_file`?

Comment (pickfire) [reply helix-term/src/keymap.rs:523]: There are a couple of reasons

- it is the same as kakoune now `gc`
- it is more aligned with view which is `zc` which have the same behavior

File: helix-view/src/document.rs [hunk 4]
```diff
@@ -344,6 +349,7 @@ impl Document {
             history: Cell::new(History::default()),
             savepoint: None,
             last_saved_revision: 0,
+            modified_since_accessed: Cell::new(false),
```
Comment (archseer) [inline helix-view/src/document.rs:352]: Why does this need to be a `Cell`? `std::mem::take()` should work just fine

Comment (pickfire) [reply helix-view/src/document.rs:352]: Thanks I forgot about `mem::take`.

File: helix-term/src/keymap.rs (hunk 3 above)
Comment (Omnikar) [reply helix-term/src/keymap.rs]: Not sure if we'd want to put this in the default keymap but I'd probably remap `goto_last_modified_file` to `gM`.

**Pull request merged**